### PR TITLE
feat(user-profile): Me-UserProfile endpoint 2.0

### PR DESCRIPTION
**Pull request by GunnlaugurG**
## What

User profile 2.0. Setup for new user profile backend, with one endpoint that returns data only from our user-profile database. 
The endpoint returns the user-profile for the current user, so if a user "A" is acting as user "B", the user-profile for user "B" will be returned. 

## Why

Se we can start querying from our database and eventually cut communications with the AdvaniaDB for user-profiles.
User-profile 2.0 will follow updated API design guide

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by rafnarnason — APPROVED** [review]
lgtm

File: apps/services/user-profile/src/app/v2/me-user-profile.controller.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { Documentation } from '@island.is/nest/swagger'
+import { UserProfileService } from './user-profile.service'
+import { UserProfileDto } from './dto/user-profileDto'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/me-user-profile.controller.ts:15]: ~We haven't added ESLint order but we are going to one day, until then we try to manually sort the orders like:

```
import { Controller, Get, UseGuards } from '@nestjs/common'
import { ApiSecurity, ApiTags } from '@nestjs/swagger'

import {
  CurrentUser,
  IdsUserGuard,
  Scopes,
  ScopesGuard,
} from '@island.is/auth-nest-tools'
import type { User } from '@island.is/auth-nest-tools'
import { UserProfileScope } from '@island.is/auth/scopes'
import { Audit } from '@island.is/nest/audit'
import { Documentation } from '@island.is/nest/swagger'

import { UserProfile } from '../user-profile/userProfile.model'
import { UserProfileDto } from './dto/user-profileDto'
import { UserProfileService } from './user-profile.service'
```

**Review by saevarma — CHANGES_REQUESTED** [review]
Nice job 🚀

Caught few things that we need to improve before merging this.

File: apps/services/user-profile/src/app/v2/dto/user-profileDto.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { IsBoolean, IsEnum, IsOptional, IsString } from 'class-validator'
+import { ApiPropertyOptional } from '@nestjs/swagger'
+
+import { DataStatus } from '../../user-profile/types/dataStatusTypes'
+
+export class UserProfileDto {
+  @ApiPropertyOptional()
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/dto/user-profileDto.ts:7]: If property is not optional it should rather use `@ApiProperty()`.
With the new strategy of returning an empty profile we need to add some properties. We will discuss this offline.

**Review by saevarma — COMMENTED** [review]
Few additional comments, did you miss few of my previous comments?

**Comment by GunnlaugurG** [comment]
> Few additional comments, did you miss few of my previous comments?

@saevarma yes it looks like i forgot a few, they where hidden under a collapsed view :D Will add them and the new comments

File: apps/services/user-profile/src/app/v2/dto/user-profileDto.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsEnum(DataStatus)
+  readonly mobileStatus?: DataStatus
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsEnum(DataStatus)
+  readonly emailStatus?: DataStatus
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/dto/user-profileDto.ts:35]: I think we should just remove these fields from the DTO and I would actually want to remove them from the model and DB table in a later task.

File: apps/services/user-profile/src/app/v2/dto/user-profileDto.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
+  readonly mobilePhoneNumber?: string
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsEnum(DataType.ENUM('en', 'is'))
+  readonly locale?: string
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/dto/user-profileDto.ts:25]: We should use the locale enum in `src/app/user-profile/types`
```suggestion
  @ApiPropertyOptional({
      enum: Locale,
      enumName: 'Locale',
  })
  @IsOptional()
  @IsEnum(Locale)
  readonly locale?: Locale
```

File: apps/services/user-profile/src/app/v2/dto/user-profileDto.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import { DataType } from 'sequelize-typescript'
+
+export class UserProfileDto {
+  @ApiProperty()
+  @IsString()
+  readonly nationalId: string
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/dto/user-profileDto.ts:14]: `class-validator` contains `IsEmail` decorator that we can use instead to validate email syntax.

**Review by saevarma — COMMENTED** [review]
These should be the final comments 💪

File: apps/services/user-profile/src/app/v2/dto/user-profileDto.ts (hunk 4 above)
Comment (GunnlaugurG) [reply apps/services/user-profile/src/app/v2/dto/user-profileDto.ts:25]: I removed the profilePicture as well, as we discussed offline

**Review by saevarma — APPROVED** [review]
Nice first step to user profiles v2!

**Pull request merged**